app/frontend/src/features/profile/edit/EditHostingPreference.test.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  render,
+  screen,
+  waitFor,
+  queryByAttribute,
+  fireEvent,
+  act,
+} from "@testing-library/react";
+import { Route, Switch } from "react-router-dom";
+import { addDefaultUser, MockedService } from "../../../test/utils";
+import wrapper from "test/hookWrapper";
+import EditHostingPreference from "./EditHostingPreference";
+import { getUser } from "test/serviceMockDefaults";
+import { service } from "service";
+import { profileRoute } from "routes";
+
+const MockProfilePage = "Mock Profile Page";
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const MockProfileComponent = () => <div>{MockProfilePage}</div>;
+
+describe("EditHostingPreference", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    addDefaultUser();
+  });
+
+  it("should redirect to Profile route after successful update", async () => {
+    const { getByText } = render(
+      <Switch>
+        <Route path={profileRoute}>
+          <MockProfileComponent />
+        </Route>
+        <Route>
+          <EditHostingPreference />
+        </Route>
+      </Switch>,
+      { wrapper }
+    );
+    const saveButton = await waitFor(() => getByText(/save/i));
+    const rightClick = { button: 2 };
+    fireEvent.click(saveButton, rightClick);
+    const profilePage = await waitFor(() => screen.getByText(MockProfilePage));
+    expect(profilePage).toBeDefined();
```

Comment:
```suggestion
    expect(await screen.findByText(mockProfilePage)).toBeInTheDocument();
```
Prefer using assertions from [`@testing-library/jest-dom`](https://github.com/testing-library/jest-dom) over the primitive jest ones since [they give better errors](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-assertion).